### PR TITLE
LibWeb: Fix Array OOBs in the HTMLTokenizer

### PR DESCRIPTION
**Pull request by stelar7**
Accessing last() if there are no elements makes the browser crash :^)

### CODE REVIEW

**Review by linusg — COMMENTED** [review]
> Accessing last() if there are no elements makes the browser crash :^)

The browser exclusively uses OOPWV these days, why would it crash?

**Comment by stelar7** [comment]
> The browser exclusively uses OOPWV these days, why would it crash?

I ment WebContent.. (Even if its not the browser, its pretty much the same imo 🤷)

**Comment by linusg** [comment]
It's absolutely not the same. Please fix the commit message.

File: Userland/Libraries/LibWeb/HTML/Parser/HTMLTokenizer.cpp [hunk 1]
```diff
@@ -210,16 +210,18 @@ Optional<u32> HTMLTokenizer::next_code_point()
...
-            m_source_positions.last().column = 0;
-            m_source_positions.last().line++;
-        } else {
+            if (!m_source_positions.is_empty()) {
+                m_source_positions.last().column = 0;
+                m_source_positions.last().line++;
+            }
+        } else if (!m_source_positions.is_empty())
             m_source_positions.last().column++;
-        }
```
Comment (linusg) [inline Userland/Libraries/LibWeb/HTML/Parser/HTMLTokenizer.cpp:222]: Please put this back, branches need curly braces if any of the sibling branches have them.

**Pull request merged**